packages/react/src/components/TreeView/TreeView-test.js

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { mount } from 'enzyme';
+import { Document16, Folder16 } from '@carbon/icons-react';
+import { settings } from 'carbon-components';
+import TreeView, { TreeNode } from './';
+
+const { prefix } = settings;
+
+describe('TreeView', () => {
+  let wrapper;
+  let onTreeSelect;
+  let onNodeSelect;
+
+  beforeEach(() => {
+    onTreeSelect = jest.fn();
+    onNodeSelect = jest.fn();
+    wrapper = mount(
+      <TreeView label="Tree view" selected={['1']} onSelect={onTreeSelect}>
+        <TreeNode id="1" value="1" label="1" />
+        <TreeNode id="2" value="2" label="2" onSelect={onNodeSelect} />
+        <TreeNode id="5" value="5" label="5" isExpanded>
+          <TreeNode id="5-1" value="5-1" label="5-1" />
+          <TreeNode id="5-2" value="5-2" label="5-2" />
+          <TreeNode id="5-3" value="5-3" label="5-3" isExpanded>
+            <TreeNode id="5-4" value="5-4" label="5-4" />
+          </TreeNode>
+        </TreeNode>
+      </TreeView>
+    );
+  });
+
+  it('should render', () => {
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('should render with icons', () => {
+    wrapper = mount(
+      <TreeView label="Tree view" selected={['1']}>
+        <TreeNode renderIcon={Document16} id="1" value="1" label="1" />
+        <TreeNode renderIcon={Document16} id="2" value="2" label="2" />
+        <TreeNode renderIcon={Folder16} id="5" value="5" label="5" isExpanded>
+          <TreeNode renderIcon={Document16} id="5-1" value="5-1" label="5-1" />
+          <TreeNode renderIcon={Document16} id="5-2" value="5-2" label="5-2" />
+          <TreeNode
+            renderIcon={Folder16}
+            id="5-3"
+            value="5-3"
+            label="5-3"
```

Comment:
Will there be anyway to provide a label that is not a string value? Or is that opening things up for it to be inaccessible?

Reply:
it is already possible to provide non-string label values, unless I am misunderstanding your question here